src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/TagHelperResolutionResultComparer.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    internal sealed class TagHelperResolutionResultComparer : IEqualityComparer<TagHelperResolutionResult>
+    {
+        internal static readonly TagHelperResolutionResultComparer Default = new TagHelperResolutionResultComparer();
+
+        public bool Equals(TagHelperResolutionResult x, TagHelperResolutionResult y)
+        {
+            return x.Descriptors.SequenceEqual(y.Descriptors, TagHelperDescriptorComparer.Default) &&
```

Comment:
Need to handle null case. It should be true if they are both null. Right now this would throw.

Reply:
Added https://github.com/dotnet/aspnetcore-tooling/pull/2217/commits/56d1919dc743d9cc85b8b87d09220ce5bdf9b15f#diff-47a48387d941d7fe0445baa8f045e7c7R18-R26